test/libsolidity/semanticTests/viaYul/array_storage_length_access.sol

```diff
@@ -1,19 +1,23 @@
 contract C {
-	uint[] storageArray;
-	function set_get_length(uint256 len) public returns (uint256)
-	{
-		storageArray.length = len;
-		return storageArray.length;
-	}
-
+    uint[] storageArray;
+    function set_get_length(uint256 len) public returns (uint256)
+    {
+        while(storageArray.length < len)
+            storageArray.push();
+        while(storageArray.length > len)
+            storageArray.pop();
+        return storageArray.length;
+    }
 }
 // ====
 // compileViaYul: true
+// EVMVersion: >=petersburg
 // ----
 // set_get_length(uint256): 0 -> 0
 // set_get_length(uint256): 1 -> 1
 // set_get_length(uint256): 10 -> 10
 // set_get_length(uint256): 20 -> 20
 // set_get_length(uint256): 0 -> 0
 // set_get_length(uint256): 0xFF -> 0xFF
-// set_get_length(uint256): 0xFFFF -> 0xFFFF
+// set_get_length(uint256): 0xFFF -> 0xFFF
+// set_get_length(uint256): 0xFFFF -> FAILURE # Out-of-gas #
```

Comment:
Why does it run out of gas?

Reply:
The yul code generated for `push()` seems to have higher gas costs than writing to `length`. Trying to narrow that down...

Reply:
I would rather assume it's an infinite loop...

Reply:
@erak Any updates here?

Reply:
@leonardoalt Yes. We're simply running out of gas because the loop is of course more expensive than a simple assignment. The same happens independently from `push` by just incrementing a counter of type `uint` in the body of the loop.

The above can be run ~9500 times on emvc with our current EVMHost settings before there is no gas left any more. I think we can keep the changes to the expectations, can't we?

Reply:
Sure, I think that's fine. Just good to know the reason

Reply:
Oh I'm sorry! I totally read this the wrong way. This is exactly the behaviour we wanted to enforce by this change...